gnomad_qc/v3/create_release/make_var_annot_hists.py

```diff
@@ -1,61 +1,164 @@
 import argparse
+import json
+import logging
 
 import hail as hl
-from gnomad.utils.annotations import (ANNOTATIONS_HISTS,
-                                      create_frequency_bins_expr,
+from gnomad.resources.resource_utils import DataException
+from gnomad.utils.annotations import (create_frequency_bins_expr,
                                       get_annotations_hists)
-from gnomad.utils.slack import try_slack
-
+from gnomad.utils.file_utils import file_exists
+from gnomad.utils.slack import slack_notifications
+from gnomad_qc.slack_creds import slack_token
 from gnomad_qc.v3.resources.constants import CURRENT_RELEASE
-from gnomad_qc.v3.resources.release import (qual_hists_json_path,
+from gnomad_qc.v3.resources.release import (annotation_hists_path,
+                                            qual_hists_json_path,
                                             release_ht_path)
 
 
+logging.basicConfig(
+    format="%(asctime)s (%(name)s %(lineno)s): %(message)s",
+    datefmt="%m/%d/%Y %I:%M:%S %p",
+)
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+
+LOG10_ANNOTATIONS = ["AS_VarDP", "QUALapprox", "AS_QUALapprox"]
+"""
+List of annotations to log scale when creating histograms. 
+"""
+
+
+def create_frequency_bins_expr_inbreeding(
+        AF: hl.expr.NumericExpression
+) -> hl.expr.StringExpression:
+    """
+    Creates bins for frequencies in preparation for aggregating QUAL by frequency bin.
+
+    Uses bins of < 0.0005 and >= 0.0005
+
+    NOTE: Frequencies should be frequencies from raw data.
+    Used when creating site quality distribution json files.
+
+    :param AC: Field in input that contains the allele count information
+    :param AF: Field in input that contains the allele frequency information
+    :return: Expression containing bin name
+    :rtype: hl.expr.StringExpression
+    """
+    bin_expr = (
+        hl.case()
+            .when(AF < 0.0005, "under_0.0005")
+            .when((AF >= 0.0005) & (AF <= 1), "over_0.0005")
+            .default(hl.null(hl.tstr))
+    )
+    return bin_expr
+
+
 def main(args):
     hl.init(default_reference='GRCh38', log='/variant_histograms.log')
 
-    ht = hl.read_table(release_ht_path())
+    logger.info("Loading ANNOTATIONS_HISTS dictionary...")
+    if not file_exists(annotation_hists_path()):
+        raise DataException(
```

Comment:
I think it's preferable to stick with `DataException` or some other user-defined exception class here since this is wrapping both `hl.hadoop_exists` and `os.path.exists`.

Not sure that OSError (and by extension FileNotFoundError) would be appropriate since those are defined as being raised by "system functions"
> This exception is raised when a system function returns a system-related error, including I/O failures such as “file not found” or “disk full” (not for illegal argument types or other incidental errors).
> -- https://docs.python.org/3/library/exceptions.html#OSError

We could define some more specific exception classes in gnomAD, including a "not found" error class.

Reply:
Thanks for explaining as always Nick! Please ignore my suggestion Julia!

Reply:
Then again... I couldn't really find any guidance on this. Hail uses a lot of OSError subclasses, such as FileNotFoundError, in their code.

¯\\_(ツ)_/¯

Reply:
I'm fine with either, FileNotFoundError does make more sense though

Reply:
@mike-w-wilson what is your preference?

Reply:
OK, leaving as is, it won't impact things overall as there is still na error, but we can cleanup in the future if we would rather something else for our standards